plugins/processors/awsapplicationsignals/internal/resolver/ecs.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package resolver
+
+import (
+	"context"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	semconv "go.opentelemetry.io/collector/semconv/v1.22.0"
+
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/common"
+	attr "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/internal/attributes"
+)
+
+type ecsResourceAttributesResolver struct {
+	resourceAttributesResolver
+	clusterName string
+}
+
+func (e *ecsResourceAttributesResolver) Process(attributes, resourceAttributes pcommon.Map) error {
+	for attrKey, mappingKey := range e.attributeMap {
+		if val, ok := resourceAttributes.Get(attrKey); ok {
+			attributes.PutStr(mappingKey, val.Str())
+		}
+	}
+
+	clusterName := e.clusterName
+	clusterNameOverwrite, taskId := getECSResourcesFromResourceAttributes(resourceAttributes)
+	if clusterNameOverwrite != "" {
+		clusterName = clusterNameOverwrite
+	}
+	attributes.PutStr(attr.AWSLocalEnvironment, e.getLocalEnvironment(attributes, resourceAttributes, clusterName))
+	attributes.PutStr(common.AttributePlatformType, e.platformType)
+
+	attributes.PutStr(attr.AWSECSClusterName, clusterName)
+	if taskId != "" {
+		attributes.PutStr(attr.AWSECSTaskID, taskId)
+	}
+	return nil
+}
+
+func (e *ecsResourceAttributesResolver) getLocalEnvironment(attributes pcommon.Map, resourceAttributes pcommon.Map, clusterName string) string {
+	if val, ok := attributes.Get(attr.AWSLocalEnvironment); ok {
+		return val.Str()
+	}
+	if val, found := resourceAttributes.Get(attr.AWSHostedInEnvironment); found {
+		return val.Str()
+	}
+	if clusterName != "" {
+		return generateLocalEnvironment(e.defaultEnvPrefix, clusterName)
+	}
+	return generateLocalEnvironment(e.defaultEnvPrefix, AttributeEnvironmentDefault)
+}
+
+func (e *ecsResourceAttributesResolver) Stop(ctx context.Context) error {
+	return nil
+}
+
+func newECSResourceAttributesResolver(defaultEnvPrefix string, clusterName string) *ecsResourceAttributesResolver {
+	return &ecsResourceAttributesResolver{
+		resourceAttributesResolver: resourceAttributesResolver{
+			defaultEnvPrefix: defaultEnvPrefix,
+			platformType:     AttributePlatformECS,
+			attributeMap:     DefaultInheritedAttributes,
+		},
+		clusterName: clusterName,
+	}
+}
+
+func getECSResourcesFromResourceAttributes(resourceAttributes pcommon.Map) (clusterName, taskId string) {
+	if clusterAttr, ok := resourceAttributes.Get(semconv.AttributeAWSECSClusterARN); ok {
+		parts := strings.Split(clusterAttr.Str(), "/")
+		clusterName = parts[len(parts)-1]
+	}
+	if taskAttr, ok := resourceAttributes.Get(semconv.AttributeAWSECSTaskARN); ok {
```

Comment:
If we already got the cluser name from the Cluster ARN, why are we trying to parse this attribute again?

Reply:
The cluster name is only set through `aws.ecs.task.arn` if it's empty, meaning the `aws.ecs.cluster.arn` doesn't exist. I don't think we would ever hit the logic, but I think it will be nice to have it as a protection. `aws.ecs.task.arn` is mainly used to extract task id.